xbmc/dialogs/GUIDialogVolumeBar.cpp

```diff
@@ -27,7 +28,9 @@ CGUIDialogVolumeBar::CGUIDialogVolumeBar(void)
   : CGUIDialog(WINDOW_DIALOG_VOLUME_BAR, "DialogVolumeBar.xml", DialogModalityType::MODELESS)
 {
   m_loadType = LOAD_ON_GUI_INIT;
-  SetAutoClose(VOLUME_BAR_DISPLAY_TIME);
+
+  if (g_application.IsMuted() || g_application.GetVolume(false) <= VOLUME_MINIMUM)
+    Open();
```

Comment:
>previously the visibility of the mute icon was toggled by checking the volume constantly:
https://github.com/xbmc/xbmc/pull/8875/files#diff-740af616a0870190df119e60bb08e0edL33
which is pretty meh i think...

Totally agree. see my comment above where to put the check.